libsolidity/LValue.h

```diff
@@ -0,0 +1,112 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author Christian <c@ethdev.com>
+ * @date 2015
+ * LValues for use in the expresison compiler.
+ */
+
+#pragma once
+
+#include <memory>
+#include <libevmcore/SourceLocation.h>
+
+namespace dev
+{
+namespace solidity
+{
+
+class Declaration;
+class Type;
+class CompilerContext;
+
+/**
+ * Abstract class used to retrieve, delete and store data in lvalues/variables.
+ */
+class LValue
+{
+protected:
+	LValue(CompilerContext& _compilerContext, std::shared_ptr<Type const> const& _dataType):
+		m_context(_compilerContext), m_dataType(_dataType) {}
+
+public:
+	/// @returns true if this lvalue reference type occupies a slot on the stack.
+	virtual bool storesReferenceOnStack() const = 0;
+	/// Copies the value of the current lvalue to the top of the stack and, if @a _remove is true,
+	/// also removes the reference from the stack.
+	/// @a _location source location of the current expression, used for error reporting.
+	virtual void retrieveValue(SourceLocation const& _location, bool _remove = false) const = 0;
+	/// Moves a value from the stack to the lvalue. Removes the value if @a _move is true.
+	/// @a _location is the source location of the expression that caused this operation.
+	/// Stack pre: value [lvalue_ref]
+	/// Stack post if !_move: value_of(lvalue_ref)
+	virtual void storeValue(Type const& _sourceType,
+		SourceLocation const& _location = SourceLocation(), bool _move = false) const = 0;
```

Comment:
I think since you broke one argument to a new line maybe you should break them all?